source/common/grpc/common.cc

```diff
@@ -26,6 +26,23 @@
 namespace Envoy {
 namespace Grpc {
 
+Common::Common(Stats::SymbolTable& symbol_table)
+    : symbol_table_(symbol_table), stat_name_pool_(symbol_table),
+      grpc_(stat_name_pool_.add("grpc")), grpc_web_(stat_name_pool_.add("grpc-web")),
+      success_(stat_name_pool_.add("success")), failure_(stat_name_pool_.add("failure")),
+      total_(stat_name_pool_.add("total")), zero_(stat_name_pool_.add("0")) {}
+
+Stats::StatName Common::makeDynamicStatName(absl::string_view name) {
+  Thread::LockGuard lock(mutex_);
```

Comment:
Is this the hot path you are worried about? Maybe add a TODO/comment here so that we can watch out for it in perf traces?